src/treeview_switches.ml

```diff
@@ -101,22 +101,31 @@ module Command = struct
         | Package _ ->
           Promise.return
           @@ show_message `Warn "The selected item is not an opam switch."
-        | Switch (opam, switch) -> (
+        | Switch (opam, switch) ->
           let open Promise.Syntax in
-          Sandbox.focus_on_package_command ();
-          let+ result = Opam.switch_remove opam switch |> Cmd.output in
-          match result with
-          | Error err -> show_message `Error "%s" err
-          | Ok _ ->
-            let (_ : Ojs.t option Promise.t) =
-              Vscode.Commands.executeCommand
-                ~command:Extension_consts.Commands.refresh_switches ~args:[]
-            in
-            let (_ : Ojs.t option Promise.t) =
-              Vscode.Commands.executeCommand
-                ~command:Extension_consts.Commands.refresh_sandbox ~args:[]
-            in
-            show_message `Info "The switch has been removed successfully." )
+          let* confirmed =
+            ask_confirmation
```

Comment:
What do you think of a function `with_confirmation` that takes a `unit -> unit Promise.t` and executes it only if the user confirmed it?

Reply:
Sounds good.